pkg/search/package.go

```diff
@@ -0,0 +1,49 @@
+package search
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/SUSE/connect-ng/pkg/connection"
+	"github.com/SUSE/connect-ng/pkg/registration"
+)
+
+// The result as given from the /package_search API.
+type SearchPackageResult struct {
+	ID       int                    `json:"id"`
+	Name     string                 `json:"name"`
+	Arch     string                 `json:"arch"`
+	Version  string                 `json:"version"`
+	Release  string                 `json:"release"`
+	Products []registration.Product `json:"products"`
+}
+
+// Performs a package search request for the given search `query`. It also
+// expects the triplet identifier for the product to be used as a `base`.
+func Package(conn connection.Connection, query, base string) ([]SearchPackageResult, error) {
```

Comment:
Can you add unit tests here? I think we should have everything tested in the library.